Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -512,6 +513,31 @@ private extension ProductStore {
         }
         storageProduct.addToTags(NSOrderedSet(array: storageTags))
     }
+
+    /// Updates, inserts, or prunes the provided StorageProduct's downloadable files using the provided read-only Product's downloadable files
+    ///
+    func handleProductDownloadableFiles(_ readOnlyProduct: Networking.Product, _ storageProduct: Storage.Product, _ storage: StorageType) {
+
+        // Removes all the downloadable files first.
+        storageProduct.downloadableFilesArray.forEach { existingStorageDownloadableFile in
+            storage.deleteObject(existingStorageDownloadableFile)
+            storageProduct.removeFromDownloads(existingStorageDownloadableFile)
+        }
```

Comment:
[Q] If we are deleting all files for a product here, wouldn't the `loadProductDownloadableFile` in line 530 always be nil? 🤔

Reply:
Yes, you are right. But for safety, I added that block.
Though, now it seems redundant to have that. Removed that block.
Thanks for the great insight and for pointing that out. 🙂